sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -217,6 +217,18 @@ class CodegenContext {
     splitExpressions(expressions = initCodes, funcName = "init", arguments = Nil)
   }
 
+  /**
+   * Return true if a given variable has been described as a global variable
+   */
+  def isDeclaredMutableState(varName: String): Boolean = {
+    val j = varName.indexOf("[")
```

Comment:
we don't need to deal with `[]` here, using them as parameter name will fail to compile.

Reply:
I see.